pkg/network/encoding/encoding.go

```diff
@@ -58,30 +53,51 @@ func GetUnmarshaler(ctype string) Unmarshaler {
 	return jSerializer
 }
 
-func modelConnections(conns *network.Connections) *model.Connections {
-	cfgOnce.Do(func() {
-		agentCfg = &model.AgentConfiguration{
+// ConnectionsModeler contains all the necessary structs for modeling a connection.
+type ConnectionsModeler struct {
+	httpEncoder                 *httpEncoder
+	http2Encoder                *http2Encoder
+	kafkaEncoder                *kafkaEncoder
+	connTelemetryMap            map[string]int64
+	compilationTelemetryByAsset map[string]*model.RuntimeCompilationTelemetry
+	kernelHeaderFetchResult     model.KernelHeaderFetchResult
+	coreTelemetryByAsset        map[string]model.COREResult
+	agentCfg                    *model.AgentConfiguration
+	prebuiltEBPFAssets          []string
+	batchIndex                  int
+}
+
+// InitConnectionsModeler initialize the connection modeler with the encoders, telemetry, and agent configuration of
+// the current connections.
+func InitConnectionsModeler(conns *network.Connections) *ConnectionsModeler {
+	return &ConnectionsModeler{
+		httpEncoder:                 newHTTPEncoder(conns.HTTP),
+		http2Encoder:                newHTTP2Encoder(conns.HTTP2),
+		kafkaEncoder:                newKafkaEncoder(conns.Kafka),
+		connTelemetryMap:            FormatConnectionTelemetry(conns.ConnTelemetry),
+		compilationTelemetryByAsset: FormatCompilationTelemetry(conns.CompilationTelemetryByAsset),
+		kernelHeaderFetchResult:     model.KernelHeaderFetchResult(conns.KernelHeaderFetchResult),
+		coreTelemetryByAsset:        FormatCORETelemetry(conns.CORETelemetryByAsset),
+		prebuiltEBPFAssets:          conns.PrebuiltAssets,
+		agentCfg: &model.AgentConfiguration{
 			NpmEnabled: config.SystemProbe.GetBool("network_config.enabled"),
 			UsmEnabled: config.SystemProbe.GetBool("service_monitoring_config.enabled"),
 			DsmEnabled: config.SystemProbe.GetBool("data_streams_config.enabled"),
-		}
-	})
+		},
+	}
+}
 
+// ModelConnections returns network connections after modeling for all supported types of traffic.
+func (c *ConnectionsModeler) ModelConnections(conns *network.Connections) *model.Connections {
 	agentConns := make([]*model.Connection, len(conns.Conns))
 	routeIndex := make(map[string]RouteIdx)
-	httpEncoder := newHTTPEncoder(conns.HTTP)
-	defer httpEncoder.Close()
```

Comment:
Does `ConnectionsModeler` need a `Close` function so it can cleanup all the encoders?